mkosi.md

```diff
@@ -287,30 +287,34 @@ a boolean argument: either "1", "yes", or "true" to enable, or "0",
 
 `Distribution=`
 
-: Matches against the configured distribution. Multiple distributions may
-  be specified, separated by spaces. If multiple distributions are specified,
-  the condition is satisfied if the current distribution equals any of the
-  specified distributions.
+: Matches against the configured distribution. Multiple distributions may be
+  specified, separated by spaces. A single distribution match can be negated by
+  preceding the value with `!`. If multiple distributions are specified, the
+  condition is satisfied if the current distribution equals any of the specified
+  distributions.
```

Comment:
Does this make sense? If I do `Distribution=centos !fedora` the `!fedora` adds zero value, there's only value if all the individual distros are prefixed with`!` but in that case, a single `!` at the front of the condition that negates the full result will also do the trick. In other words, instead of allowing negation of individual list items, should we instead just support negating the full result of the condition?

Reply:
I wondered about this myself and went for negating single items since that was easiest to implement, since it can be done at each step and we don't have to somehow track whether we're at the end and negate then. 

I agree this can have unforeseen edge cases and some are covered in the tests, but I figured it would cover most usage, since people will most often match a single thing or want to exclude a single thing, e.g. "Include this package for everything but Fedora (with everything probably being some RHEL derivative)" or "Include this package for this old release, but not in newer ones."

Reply:
In other words, I expect usage to look like this: You're a Fedora/RHEL shop, than the common match will look like either of these
```
Distribution=centos
Distribution=!fedora
```
which is why I think that usually negating a single thing is already negating the whole thing. 

Also, if only the whole thing can be negated we just switch the situation around and to solve that we'd need parentheses (which in a sense we already have in the sense of adding another file).

Reply:
I think negating the whole thing would fit better.
If I want to exclude for example centos and fedora for some reason, this would not work with single negation, since
`!fedora || !centos = !(fedora && centos)`, so the condition is always true.
If the whole thing is negated, this would be `!(fedora || centos) = !fedora && !centos` and this exclusion would work.